plan4past/compiler.py

```diff
@@ -202,3 +215,333 @@ def _update_domain_actions_det(
                 )
             )
     return new_actions
+
+
+class ProblemUnsolvableException(Exception):
+    def __init__(self, *args: object) -> None:
+        super().__init__(*args)
```

Comment:
not super convinced by this; this looks like a semantic error, not a syntactic one. Therefore, the compiler should not care (perhaps it could be a warning).

Compare this with PDDL problems with unsolvable goals. What do planners do in those cases: do they return error, or just return "no plan found"?